### PR TITLE
refactor: Remove nickname parsing

### PR DESCRIPTION
**Pull request by Jiralite**
**Please describe the changes this PR makes and why it should be merged:**

- https://github.com/discord/discord-api-docs/issues/4734#issuecomment-1086156692
- https://github.com/discord/discord-api-docs/commit/a8f7638e79391bb70e15996d1d88488a3b6364b2

**Status and versioning classification:**
- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

File: packages/discord.js/src/structures/MessageMentions.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ class MessageMentions {
    * @type {RegExp}
    * @memberof MessageMentions
    */
-  static UsersPattern = /<@!?(\d{17,19})>/g;
+  static UsersPattern = /<@(\d{17,19})>/g;
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/structures/MessageMentions.js:22]: `<@!id>` is also a valid user mention. The issue is it won't show nickname/username but instead both `<@id>` and `<@!id>` will show the same result.

File: packages/discord.js/src/util/Util.js [hunk 2]
```diff
@@ -526,7 +526,7 @@ class Util extends null {
    */
   static cleanContent(str, channel) {
     str = str
-      .replace(/<@!?[0-9]+>/g, input => {
+      .replace(/<@[0-9]+>/g, input => {
```
Comment (imranbarbhuiya) [inline packages/discord.js/src/util/Util.js:529]: same

File: packages/discord.js/src/structures/MessageMentions.js (hunk 1 above)
Comment (ImRodry) [reply packages/discord.js/src/structures/MessageMentions.js:22]: They made it seem like they were gonna remove it though https://github.com/discord/discord-api-docs/issues/4734#issuecomment-1086156692

Comment (splatterxl) [reply packages/discord.js/src/structures/MessageMentions.js:22]: No, they're just not going to send it from the client. Making an extremely breaking change like that with such short notice is not really what Discord would do.

**Comment by ImRodry** [comment]
Should also change this
https://github.com/discordjs/discord.js/blob/ac4bc3a6c81f7b86bc5ec2b05069122a601224e7/packages/discord.js/src/structures/GuildMember.js#L431

**Comment by almostSouji** [comment]
Discord is removing it, but does it hurt to just keep it in the parsing section for legacy reasons (parsing messages where this was still the case)? I do agree with removing it from #toString, so d.js doesn't send it anymore, but do not see any benefit in removing it from the parsing regex

**Comment by vladfrangu** [comment]
Agreed with @almostSouji, let's remove it just from toString() (aka make toString() call this.user.toString() :megu:)

**Review by kyranet — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**